### ISSUE TITLE
SPInstall: Incorrectly detects that SharePoint is already installed

### ISSUE THREAD

**Issue by jensotto** [PowerShell/SharePointDsc#1099]
#### Details of the scenario you tried and the problem that is occurring
Uninstall SharePoint (2019) from a server and use DSC to reconfigure the same server.
SPInstall resource incorrectly detects that SharePoint is already installed and skips installation.

#### Verbose logs showing the problem
Did not keep the log for this, but is identical to the output when SharePoint is installed.

#### Suggested solution to the issue
This happens because the filterscript in Get-Resource checks for installed items CONTAINING the string "Microsoft SharePoint Server 2013", "Microsoft SharePoint Server 2016", or "Microsoft SharePoint Server 2019". After uninstallation of SharePoint there could be leftover components that contains this text, even for newly installed nodes in a new farm.
Change the filterscript that is responsible for detecting if SharePoint is installed so that it searches for an exact string, or some other value. It should also only check for the version that is to be installed and not any version of SharePoint (2013/2016/2019).

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
            SPInstall SharePointInstallation
            {
                IsSingleInstance = 'Yes'
                BinaryDir = $ConfigurationData.NonNodeData.SPInstallationBinaryPath;
                ProductKey = $SPProductKey;
                Ensure = "Present";
                DataPath = $ConfigurationData.NonNodeData.SharePointDataLocation;
                PSDscRunAsCredential = $SPSetupCredential;
            }
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2016 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 14393.2969.amd64fre.rs1_release.190503-1820
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of SharePoint that is used (e.g. SharePoint 2016)
SharePoint 2019

#### Version and build of PowerShell the target node is running
Name                           Value
----                           -----
PSVersion                      5.1.14393.2969
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.2969
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

#### Version of the DSC module that was used ('dev' if using current dev branch)
3.5.0.0

**Comment by jensotto** [comment]
I'm not sure if "Microsoft SharePoint Server <version of SharePoint>" is possible to be used for an exact match for all different version and editions of SharePoint. It seems to work for SharePoint 2019, but we need to check if it can be used for all other editions too (Foundation/Enterprise/any other different editions that might exists).

**Comment by ykuijs** [comment]
@jensotto Did you have a language pack installed on this machine??

Looks like the naming of the language pack has changed in SP2019, now including "Microsoft SharePoint Server 2019" in the name where in SP2016 this was "Language Pack for SharePoint and Project Server 2016".

**Comment by ykuijs** [comment]
Possible solution would be to update the check to:

Current:
https://github.com/PowerShell/SharePointDsc/blob/47ef4acf75bbb06965b9a102b1b2ca793cbb79be/Modules/SharePointDsc/DSCResources/MSFT_SPInstall/MSFT_SPInstall.psm1#L94-L96

New:
```PowerShell
    $installedItems = $installedItems.DisplayName | Select-Object -Unique
    $spInstall = $installedItems | Where-Object -FilterScript {
        $_ -match "^Microsoft SharePoint Server (2013|2016|2019)$"
    }
```
This will select only the products that match the string exactly, no language packs or anything else.

**Comment by jensotto** [comment]
No, I did not have any language packs installed.
Your solution should work, but would there be issues with for example SharePoint Foundation, or SharePoint installation media in other languages than English?

**Comment by ykuijs** [comment]
As far as I know, if you install a non-English version of SharePoint you actually install the English version with the localized language pack.

**Comment by ykuijs** [comment]
This issue is now fixed in dev and will be released with v3.7
